### PR TITLE
Implement Steam Deck HIDAPI driver to enable gyroscope support (SDL 3 version)

### PR DESCRIPTION
**Pull request by mmmaisel**
## Description
This pull request implements a HIDAPI driver for the Steam Deck controller to enable gyroscope support.
The code is mostly based on the existing Steam controller HIDAPI driver.

It is a SDL3 port of the SDL2 version of this pull request which was used to test the driver in real games through the Antimicrox input mapper.

This approach works well if both, mapper and game use SDL. This includes wine/proton which use SDL under the hood.
It may not work for non-SDL native games due to the kernel quirk mentioned in #8125.

## Existing Issue(s)
Related to #8125.

### CODE REVIEW

**Comment by mmmaisel** [comment]
The CI on the branch seems to fail for similar reasons than on the main branch.

**Comment by slouken** [comment]
A few comments:
* What is the use case for this where you wouldn't have Steam available to provide controller input? Is it just to get raw access to the gyro?
* You shouldn't need the game controller mapping entry, you should instead make sure the buttons and axes line up with the gamepad API, and then the standard mapping will work fine.
* With a few adjustments this should work with all wired Steam Controllers, not just the Steam Deck controller (I think?), is there any reason you didn't include support for those?

**Comment by slouken** [comment]
BTW, you should be able to use testcontroller from the SDL test directory to validate this in SDL3.

**Comment by mmmaisel** [comment]
Thank you for your feedback and pointing me to the `testcontroller` tool, @slouken.

Currently, the main use case of this driver is adding gyro support through an
external mapper like AntimicroX to non-Steam games or Steam games the do not or
have broken Steam-Input support.
Another use case in the future may be playing non-Steam games with gyro
support (I don't know a game that does this at the moment).

I double checked the necessity of the controller mapping entry.
While everything works fine without it, the "..." key and the backside
paddles are only detected as generic buttons and are not mapped to a button or
paddles in the controller picture by the SDL3 `testcontroller` app without it.
For reference, I mapped them to SDL_GAMEPAD_BUTTON_RIGHT_PADDLE1 and similar.

The wired steam controller uses a similar but different protocol than the
steam deck controller. Main differences are the messages,
as shown in `steam/controller_structs.h`, and the absence of a watchdog message.
But you're right that this could be merged together as it is done in the Linux
kernel driver. I decided against doing it because I do not have access to a
Steam Controller to test my changes.

**Comment by slouken** [comment]
Fair enough.

File: src/joystick/SDL_gamepad_db.h [hunk 1]
```diff
@@ -727,6 +727,7 @@ static const char *s_GamepadMappings[] = {
...
     "05000000de2800000611000001000000,Steam Controller,a:b0,b:b1,back:b6,dpdown:b14,dpleft:b15,dpright:b13,dpup:b12,guide:b8,leftshoulder:b4,leftstick:b9,lefttrigger:a2,leftx:a0,lefty:a1,paddle1:b11,paddle2:b10,rightshoulder:b5,righttrigger:a3,start:b7,x:b2,y:b3,",
     "03000000de2800000512000011010000,Steam Deck,a:b3,b:b4,back:b11,dpdown:b17,dpleft:b18,dpright:b19,dpup:b16,guide:b13,leftshoulder:b7,leftstick:b14,lefttrigger:a9,leftx:a0,lefty:a1,misc1:b2,paddle1:b21,paddle2:b20,paddle3:b23,paddle4:b22,rightshoulder:b8,rightstick:b15,righttrigger:a8,rightx:a2,righty:a3,start:b12,x:b5,y:b6,",
+    "03000000de2800000512000000016800,Steam Deck,platform:Linux,crc:17f6,a:b0,b:b1,x:b2,y:b3,back:b4,guide:b5,start:b6,leftstick:b7,rightstick:b8,leftshoulder:b9,rightshoulder:b10,dpup:b11,dpdown:b12,dpleft:b13,dpright:b14,misc1:b15,paddle1:b17,paddle2:b16,paddle3:b19,paddle4:b18,leftx:a0,lefty:a1,rightx:a2,righty:a3,lefttrigger:+a4,righttrigger:a5,",
```
Comment (slouken) [inline src/joystick/SDL_gamepad_db.h:730]: The trigger binding is probably wrong, I'm guessing lefttrigger is a4 (without the `+`)
You should remove the platform: and crc: fields, they're not needed in this context.

Comment (mmmaisel) [reply src/joystick/SDL_gamepad_db.h]: I fixed the controllerdb entry as you suggested and retested it. Everything still works totally fine.

Comment (slouken) [reply src/joystick/SDL_gamepad_db.h]: Great, thanks!

**Pull request merged**